main.go

```diff
@@ -154,3 +220,197 @@ func gatekeeperNamespace(platformInfo platform.PlatformInfo) (string, error) {
 
 	return ns, nil
 }
+
+func (c *ConstraintStatusInstaller) addConstraintStatusManager(ctx context.Context, mgr manager.Manager,
+	dynamicClient *dynamic.DynamicClient, namespace string,
+) error {
+	if err := (&controllers.ConstraintStatusReconciler{
+		Client:        mgr.GetClient(),
+		ClientSet:     kubernetes.NewForConfigOrDie(mgr.GetConfig()),
+		DynamicClient: dynamicClient,
+		Log:           ctrl.Log.WithName("controllers").WithName("ConstraintPodStatus"),
+		Namespace:     namespace,
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "ConstraintPodStatus")
+		return err
+	}
+
+	if err := mgr.AddHealthzCheck("healthz", healthz.Ping); err != nil {
+		setupLog.Error(err, "unable to set up health check")
+		return err
+	}
+	if err := mgr.AddReadyzCheck("readyz", healthz.Ping); err != nil {
+		setupLog.Error(err, "unable to set up ready check")
+		return err
+	}
+	if err := mgr.Start(ctx); err != nil {
+		setupLog.Error(err, "problem running ConstraintStatus manager")
+		return err
+	}
+
+	return nil
+}
+
+func (c *ConstraintStatusInstaller) getIsInstalled() bool {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	return c.isInstalled
+}
+
+func (c *ConstraintStatusInstaller) setIsInstalled(isInstalled bool) {
+	c.mu.Lock()
+	c.isInstalled = isInstalled
+	c.mu.Unlock()
+}
+
+// Check gatekeeper auditFromCache=Automatic
+// Check constraintpodstatuses.status.gatekeeper.sh crd status has "NamesAccepted" condition that is true.
+func (c *ConstraintStatusInstaller) isApiAvailable(
```

Comment:
Based on my other comments, you may not need this method anymore. The reason being is that you no longer watch the CRD and I think controller-runtime just keeps retrying the watch request if the CRD isn't yet available.

Reply:
Actually, it does seem that you check the `Gatekeeper` object here, so this could be moved to the calling method and the rest could be removed.

Reply:
the rest? checking crd?